File: server/Domain/Models/Order.cs
namespace Domain.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;

    public class Order : EntityBase
    {
        public Order()
        {
            OrderLines = new List<OrderLine>();
        }

        [Required]
        public ICollection<OrderLine> OrderLines { get; set; }

        [Required]
        public decimal Price { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public OrderStatus OrderStatus { get; set; }
    }
}

File: client/src/interfaces/order.ts
import { IOrderLine } from 'src/interfaces/orderLine';
import { IOrderStatus } from 'src/interfaces/orderStatus';

export interface IOrder {
  id: string;
  orderLines: IOrderLine[];
  price: number;
  orderStatus: IOrderStatus;
  date: Date;
}

export interface IOrderProps {
  order: IOrder;
}
